### PR TITLE
Title: A Friendly Beginner's Guide to TypeScript

### LINKED ISSUE #1291: A Friendly Beginner's Guide to TypeScript
### Brief Summary: 
TypeScript is a programming language that extends JavaScript by adding types. It saves you time catching errors and providing fixes before you run code. This tutorial is to provide a beginner guide to a developer trying to upskill or learn TypeScript.

### Key Takeaways: 
- Understand TypeScript fundamentals
- Compiling and Configuration of TypeScript 
- Classes in Typescript
- Interface in TypeScript
- Module and Namespaces

### References: 
https://www.typescriptlang.org/docs/
https://github.com/dzharii/awesome-typescript

### PR DESCRIPTION
**Pull request by Olanetsoft**
### Article Checklist
**NOTE: (1st time contributors - submit ONLY one article to get familiar with the process, quality, and style-once published, you can then submit any number of articles.)**
#### Preliminary Checks - Formatting and Structure 
- [ ] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [ ] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [ ] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [ ] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [ ] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [ ] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [ ] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [ ] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [ ] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar Checks
- [ ] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [ ] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [ ] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

@hectorkambow here is the new clean PR. 
fixes #1291

### CODE REVIEW

**Comment by ReactifyStudio** [comment]
@hectorkambow a peer reviewer. Wishing to review this Typescript article

**Comment by zolomohan** [comment]
> @hectorkambow a peer reviewer. Wishing to review this Typescript article

Didn't notice. Sorry 😅

**Comment by kayere** [comment]
@zolomohan and @ReactifyStudio - This is a duplicate pr. I was reviewing it. 
Refer to #1311

**Comment by zolomohan** [comment]
😂😅

**Comment by ReactifyStudio** [comment]
@zolomohan you can continue with this. My PayPal chaos continues. No need to.

**Comment by zolomohan** [comment]
@Olanetsoft Be sure to mention the old PR in the duplicate PR to avoid confusion in the future.

**Comment by Olanetsoft** [comment]
@zolomohan the old PR has been closed by @kayere

File: articles/a-friendly-beginner-guide-to-typescript/index.md [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+  Orange = 1,
+  Apple = 4,
+  Banana = 7
+}
+// can be accessed by
+let c: Fruits = Fruits.Banana;
+```
+
+#### Array
+TypeScript also supports arrays.
```
Comment (kayere) [inline articles/a-friendly-beginner-guide-to-typescript/index.md:81]: Please explain what is an array.

**Review by kayere — COMMENTED** [review]
@Olanetsoft - Please make the following change and add author details.
Don't forget to pull my changes.

**Comment by Olanetsoft** [comment]
@kayere Updated Changes

**Comment by kayere** [comment]
@hectorkambow - Please give this article a final review.

**Comment by hectorkambow** [comment]
@Olanetsoft  
👍  
Great overall tutorial - please go over my final edits and changes to incorporate into future articles 🚀  

cc @kayere  please do the same 👍

**Pull request merged**